app/views/layouts/_searchbar.html.haml

```diff
@@ -23,7 +23,7 @@
       .form-group
         %button.btn.btn-default{:type => "submit"}
           %span.fa.fa-search
-        - if @lastaction == "show_list" && !session[:menu_click] && display_adv_search? && !@in_a_form
+        - if display_adv_search?
```

Comment:
Are you sure you don't alter the functionality on other pages by changing this?

Reply:
Yes I am. You can check the original lines 1 and 26, they are totally the same and there isn't any other code in between which would change values of the variables in the meantime. Or am I missing something? Also :eyes: how the structure of the whole haml looks like. Originally, if the condition in line 1 was not true, Search was not rendered at all. But it was not good as supporting displaying Advanced Search shouldn't be required for displaying regular Search. 

Also you can run MIQ with the changes in this PR and see if something unexpected happens :smirk: